File: FastMines_core/src/fmg/core/img/IMosaicAnimatedModel.cs
using Fmg.Core.Mosaic;
using Fmg.Core.Types;

namespace Fmg.Core.Img {

    public enum EMosaicRotateMode {
        /// <summary> rotate full matrix (all cells) </summary>
        fullMatrix,
        /// <summary> rotate some cells (independently of each other) </summary>
        someCells
    }

    /// <summary> Representable <see cref="EMosaic"/> as animated image </summary>
    public interface IMosaicAnimatedModel<out TImageInner>
                   : IMosaicDrawModel<TImageInner>, IAnimatedModel
        where TImageInner : class
    {

        EMosaicRotateMode RotateMode { get; set; }

        /// <summary> 0° .. +360° </summary>
        double RotateAngle { get; set; }
    }

}

File: FastMines_core/src/fmg/core/img/IMosaicAnimatedModel.java
package fmg.core.img;

import fmg.core.mosaic.IMosaicDrawModel;

/** Representable {@link fmg.core.types.EMosaic} as animated image */
@Deprecated
public interface IMosaicAnimatedModel<TImageInner>
         extends IMosaicDrawModel<TImageInner>, IAnimatedModel
 {

    public enum EMosaicRotateMode {
        /** rotate full matrix (all cells) */
        fullMatrix,
        /** rotate some cells (independently of each other) */
        someCells
    }

    EMosaicRotateMode getRotateMode();
    void setRotateMode(EMosaicRotateMode value);

    /** 0° .. +360° */
    double getRotateAngle();
    void setRotateAngle(double value);

}
